src/NServiceBus.Core/InitializableEndpoint.cs

```diff
@@ -30,8 +30,7 @@ class InitializableEndpoint
 
             RegisterContainerAdapter(container);
             RunUserRegistrations(registrations);
-
-            this.container.RegisterSingleton(this);
```

Comment:
This is a public type so technically downstreams could access it.

How about adding this to a method that we put an ObsoletEx on to make sure it gets removed in v8?

Reply:
My bad, type is internal